src/main/java/io/github/mirromutth/r2dbc/mysql/MySqlSslConfiguration.java

```diff
@@ -0,0 +1,241 @@
+/*
+ * Copyright 2018-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.github.mirromutth.r2dbc.mysql;
+
+import io.netty.handler.ssl.ClientAuth;
+import io.netty.handler.ssl.SslContext;
+import io.netty.handler.ssl.SslProvider;
+import io.netty.handler.ssl.util.InsecureTrustManagerFactory;
+import reactor.util.annotation.Nullable;
+
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.TrustManagerFactory;
+import java.net.URL;
+
+import static io.github.mirromutth.r2dbc.mysql.internal.AssertUtils.requireNonNull;
+
+/**
+ * MySQL configuration of SSL.
+ */
+public final class MySqlSslConfiguration {
+
+    private static final String DEFAULT_CERT_TYPE = "JKS";
+
+    private static final ClientAuth DEFAULT_CLIENT_AUTH = ClientAuth.NONE;
+
+    private final SslProvider sslProvider;
+
+    private final ClientAuth clientAuth;
+
+    private final String keyCertType;
+
+    private final String trustCertType;
+
+    @Nullable
+    private final String[] tlsProtocols;
```

Comment:
Make sure to decouple the array when accepting or returning it (e.g. `System.arraycopy(…)`). Alternatively, use `List` as it is easier to create a  `new ArrayList<>(original)`.

Reply:
`String[]` looks like easier to used by `SslContextBuilder.protocols(String...)`, I will try to use `System.arraycopy(…)` in `MySqlSslConfiguration.Builder.tlsProtocols(...)`.